pgmpy/factors/CPD.py

```diff
@@ -112,6 +112,27 @@ def __init__(self, variable, variable_card, values,
 
         super().__init__(variables, cardinality, values.flatten('C'))
 
+    def replace_variables(self, var1, var2):
+        new_scope = self.scope()
+        new_scope.reverse()
+        new_cardinality = self.cardinality[::-1].copy()
```

Comment:
@ankurankan @abinashpanda : I am using ```.copy()``` here. If I don't use that, my whole code goes berserk. The problem is that in line number 125, I am trying to swap two values of the numpy array ```new_cardinality```. If I don't use ```copy()``` here, the values are also swapped in ```self.cardinality()```

Is this consistent?